### PR TITLE
Move EntityTest into subclasses

### PR DESCRIPTION
**Pull request by Benestar**
By moving all the test methods from EntityTest into subclasses, we make
sure that still all tests are executed when removing the inheritance
between Item/Property and Entity. We can then start to remove tests that
test methods no longer existing while still having the old tests.

**Note:** I changed this patch fundamentally.

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Had very quick look. +0.5

File: tests/unit/Entity/EntityTest.php [hunk 1]
```diff
@@ -14,21 +14,20 @@
...
+class EntityTest extends \PHPUnit_Framework_TestCase {
 
 	/**
-	 * @since 0.1
-	 *
 	 * @return Entity
 	 */
-	protected abstract function getNewEmpty();
+	private function getNewEmpty() {
+		return new Item();
```
Comment (thiemowmde) [inline tests/unit/Entity/EntityTest.php]: Huh? This appears to be wrong.

File: tests/unit/Entity/EntityTest.php [hunk 2]
```diff
@@ -14,21 +14,20 @@
...
  *
- * @group Wikibase
- * @group WikibaseDataModel
+ * @covers Wikibase\DataModel\Entity\Entity
  *
  * @licence GNU GPL v2+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  * @author Daniel Kinzler
  */
-abstract class EntityTest extends \PHPUnit_Framework_TestCase {
```
Comment (thiemowmde) [inline tests/unit/Entity/EntityTest.php]: Why is this not abstract any more?

File: tests/unit/Entity/EntityTest.php [hunk 3]
```diff
@@ -14,21 +14,20 @@
  * @deprecated
  * This test class is to be phased out, and should not be used from outside of the component!
  *
- * @group Wikibase
- * @group WikibaseDataModel
```
Comment (thiemowmde) [inline tests/unit/Entity/EntityTest.php]: Why did you removed these?

File: tests/unit/Entity/ItemTest.php [hunk 4]
```diff
@@ -368,4 +391,18 @@ public function testNotEquals( Item $firstItem, Item $secondItem ) {
 		$this->assertFalse( $secondItem->equals( $firstItem ) );
 	}
 
+	public function testCopy() {
```
Comment (thiemowmde) [inline tests/unit/Entity/ItemTest.php]: Conflicts with #626.

File: tests/unit/Entity/EntityTest.php (hunk 2 above)
Comment (Benestar) [inline tests/unit/Entity/EntityTest.php]: We can also just remove the test in this patch already. It doesn'r really matter anyways. However, I didn't want to leave `Entity` untested and thus made the test non-abstract so that it tests a `Entity` implementation instead.

File: tests/unit/Entity/EntityTest.php (hunk 1 above)
Comment (Benestar) [inline tests/unit/Entity/EntityTest.php]: see above

File: tests/unit/Entity/EntityTest.php (hunk 2 above)
Comment (thiemowmde) [inline tests/unit/Entity/EntityTest.php]: Oh, I see. Makes sense. I confused this with the abstract `Entity` class.

File: tests/unit/Entity/EntityTest.php (hunk 3 above)
Comment (thiemowmde) [inline tests/unit/Entity/EntityTest.php]: As said in an other pull request, this is unrelated and should be done in a separate pull request. Personally I prefer to add both `@group Wikibase` and `@group WikibaseDataModel` to all tests.

**Comment by Benestar** [comment]
I re-added the `@group` annotations so maybe you can have another look into this.

File: tests/unit/Entity/PropertyTest.php [hunk 5]
```diff
@@ -229,4 +262,18 @@ public function testPropertyWithStatementsIsNotEmpty() {
 		$this->assertFalse( $property->isEmpty() );
 	}
 
+	public function testCopy() {
```
Comment (thiemowmde) [inline tests/unit/Entity/PropertyTest.php]: This conflicts heavily with what I did in #626.

File: tests/unit/Entity/PropertyTest.php [hunk 6]
```diff
@@ -229,4 +262,18 @@ public function testPropertyWithStatementsIsNotEmpty() {
...
+		$property = Property::newFromType( 'string' );
+		$property->setLabel( 'en', 'Foo' );
+
+		$copy = $property->copy();
+
+		$this->assertNotSame( $property, $copy );
+		$this->assertTrue( $property->equals( $copy ) );
```
Comment (thiemowmde) [inline tests/unit/Entity/PropertyTest.php]: The original implementation compared the id, which is still necessary.

**Comment by thiemowmde** [comment]
This copies only a few test cases, but other are now missing and not run on Property any more.
